src/main/docs/serializers.adoc

```diff
@@ -0,0 +1,265 @@
+----
+
+<1> Create a multi-tiered cache that depends on key and value serialization.
+<2> Configure a serializer for the values. The `DumbTransientStringSerializer` in this case. For the sake of simplicity
+    we have omitted key serializer. Since one is not provided explicitly, ehcache would provide default serializers to
+    perform the key serialization.
+<3> Verify that the cache/serializer works.
+
+In the previous section we demonstrated the use of a transient serializer.
+We used that serializer with a transient cache and everything works just fine.
+Now imagine what would happen if we use the same serializer with a persistent cache.
+Everything would work as long as your application is running. Once you close the cache manager or end the application 
+the data associated with the cache will be persisted so that the same data will be available on a restart.
+But there is a serious problem. The following piece of code would demonstrate that:
+
+[source,java,indent=0]
+----
+include::{sourcedir}/test/java/org/ehcache/demo/serializer/SerializersDemo.java[tag=transientSerializerBadSample]
+----
+
+<1> Create a cache configuration with persistent disk tier.
+<2> Configure the `LocalPersistenceService` for the cache manager.
+<3> Retrieve the cache.
+<4> Populate data.
+<5> Verify that everything works.
+<6> Close the cache manager.
+<7> Reinitialize the cache manager.
+<8> Retrieve the cache.
+<9> Retrieve a cached/persisted value.
+
+The above piece of code would fail in the cache creation step since the serializer provided does not meet the 2-arg 
+constructor requirement for persistent caches.
+But why does `Ehcache` enforce this requirement and fail-fast if the requirement is violated?
+What would have happened if we had proceeded with the sample code? Would it have failed? If yes, then where?
+The above piece of code would have failed in step 9 because the cache would not be able to retrieve the persisted data.
+Because the serializer that you provided would fail in retrieving that data.
+When the cache is reinitialized, the associated serializer instance is also initialized for the cache to work.
+But the newly initialized serializer would have an empty state(empty *stringIdMap* and *idStringMap* maps and the *id*
+initialized to 0).
+So when the cache tries to read a value it gets an integer value from the persistent tier as that is what got persisted.
+But using the empty state the serializer will not able to map that value to a fruit name, and so it would throw.
+That leaves the persisted data unusable.
+So what could you have done differently to make it work? The answer is simple. Persist the serializer's state as well
+and restore it when the cache is re-initialized.
+And that is exactly what persistent serializers would do.
+
+=== Persistent serializers
+
+A persistent serializer persists its state and retrieves it when reinitialized.
```

Comment:
Again, quite a long paragraph.
I would have the first two sentences as bullet points and the rest probably split in multiple paragraphs.